### PR TITLE
Doc `ipyparallel` & `mpi4py` `concurrent.futures`

### PR DESCRIPTION
**Pull request by jakirkham**
Mention `ipyparallel` and `mpi4py` also support the `concurrent.futures` API and could be used as well.

- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `black dask` / `flake8 dask` / `isort dask`

### CODE REVIEW

**Comment by ncclementi** [comment]
@jakirkham @jrbourbeau  Do you think examples for these cases would be helpful too? As a relatively new to dask person, I was going through the links and I noticed that the mpi4py case is clear, although the ipyparallel doesn't have examples for the executor.

**Comment by jakirkham** [comment]
It's a good question. There is a little bit of documentation for `ipyparallel`, but agree it is a bit sparse. Raised issue ( https://github.com/ipython/ipyparallel/issues/459 ) about expanding on this a bit with an example.

Personally think one example with Loky is probably good enough for us (and is also nice for those familiar with the scikit-learn/joblib collection of libraries). The rest of these serve as other executors one might consider based on their needs. Am sure there are probably more that I'm not thinking of that people might want to use for different reasons.

File: docs/source/scheduling.rst [hunk 1]
```diff
@@ -208,3 +208,9 @@ reusable ``ProcessPoolExecutor`` from loky_:
    from loky import get_reusable_executor
    with dask.config.set(pool=get_reusable_executor(max_workers=4)):
        x.compute()
+
+Other libraries like ipyparallel_ and mpi4py_ also supply
+``concurrent.futures.Executor`` subclasses that could be used as well.
+
+.. _ipyparallel: https://ipyparallel.readthedocs.io/en/latest/api/ipyparallel.html#ipyparallel.Client.executor
+.. _mpi4py: https://mpi4py.readthedocs.io/en/stable/mpi4py.futures.html
```
Comment (leofang) [inline docs/source/scheduling.rst:216]: We've been slowly making progress toward the next release, but for the time being I'd suggest people (especially HPC people) to always refer to the latest docs. While the stable branch (for v3.0.3) does have `mpi4py.futures` included, it missed a lot of new features we added in the past ~2 years.
```suggestion
.. _mpi4py: https://mpi4py.readthedocs.io/en/latest/mpi4py.futures.html
```

**Review by leofang — CHANGES_REQUESTED** [review]
Thanks, John! LGTM except for one nitpick.

File: docs/source/scheduling.rst (hunk 1 above)
Comment (jakirkham) [reply docs/source/scheduling.rst]: Makes sense. Updated :)

**Review by jrbourbeau — APPROVED** [review]
Thanks @jakirkham! This is in

**Comment by jakirkham** [comment]
Thanks all! 😀

**Comment by jakirkham** [comment]
FWIW here's an example using `ipyparallel`

```python
from functools import partial

import ipyparallel as ipp

import dask
import dask.array as da


c = ipp.Client()
e = c.executor()

submit = e.submit
num_workers = len(e.view)
get = partial(
    dask.local.get_async,
    submit,
    num_workers
)

with dask.config.set(scheduler=get):
    a = da.ones((1_000, 1_100), chunks=100)
    r = a.sum()
    print(r.compute())
```

Ideally we could absorb some of the boilerplate around wrapping the `submit` function to improve usability.

**Pull request merged**